pkg/controller/servicebindingrequest/annotations/resource.go

```diff
@@ -41,10 +41,10 @@ type ResourceHandler struct {
 
 // discoverRelatedResourceName returns the resource name referenced by the handler. Can return an
 // error in the case the expected information doesn't exist in the handler's resource object.
-func discoverRelatedResourceName(obj map[string]interface{}, bindingInfo *BindingInfo) (string, error) {
+func discoverRelatedResourceName(obj map[string]interface{}, bi *bindingInfo) (string, error) {
 	resourceNameValue, ok, err := unstructured.NestedFieldCopy(
 		obj,
-		strings.Split(bindingInfo.ResourceReferencePath, ".")...,
+		strings.Split(bi.ResourceReferencePath, ".")...,
 	)
 	if !ok {
 		return "", ResourceNameFieldNotFoundErr
```

Comment:
https://github.com/redhat-developer/service-binding-operator/pull/407#discussion_r418064113 needs to be addressed

Reply:
along with https://github.com/redhat-developer/service-binding-operator/pull/407#discussion_r418064449

Reply:
I believe both have been solved in 715513e20bb2440da65fce724ccdfee3b10423c6.